src/MSALWrapper/AuthFlow/Broker.cs

```diff
@@ -84,57 +77,52 @@ protected override async Task<(TokenResult, IList<Exception>)> GetTokenInnerAsyn
+
+                if (tokenResult != null)
                 {
-                    try
-                    {
-                        var tokenResult = await TaskExecutor.CompleteWithin(
-                            this.logger,
-                            this.silentAuthTimeout,
-                            "Get Token Silent",
-                            (cancellationToken) => this.pcaWrapper.GetTokenSilentAsync(this.scopes, account, cancellationToken),
-                            this.errors)
-                            .ConfigureAwait(false);
-                        tokenResult.SetSilent();
-
-                        return (tokenResult, this.errors);
-                    }
-                    catch (MsalUiRequiredException ex)
-                    {
-                        this.errors.Add(ex);
-                        this.logger.LogDebug($"Silent auth failed, re-auth is required.\n{ex.Message}");
-                        var tokenResult = await TaskExecutor.CompleteWithin(
-                            this.logger,
-                            this.interactiveAuthTimeout,
-                            "Interactive Auth",
-                            (cancellationToken) => this.pcaWrapper
-                            .WithPromptHint(this.promptHint)
-                            .GetTokenInteractiveAsync(this.scopes, account, cancellationToken),
-                            this.errors)
-                            .ConfigureAwait(false);
+                    return (tokenResult, this.errors);
+                }
+
+                Func<CancellationToken, Task<TokenResult>> getTokenInteractive = (cancellationToken) =>
+                    this.pcaWrapper
+                        .WithPromptHint(this.promptHint)
+                        .GetTokenInteractiveAsync(this.scopes, account, cancellationToken);
 
-                        return (tokenResult, this.errors);
-                    }
+                try
+                {
+                    tokenResult = await TaskExecutor.CompleteWithin(
+                        this.logger,
+                        this.interactiveAuthTimeout,
+                        $"{this.Name()} interactive auth",
+                        getTokenInteractive,
+                        this.errors).ConfigureAwait(false);
                 }
                 catch (MsalUiRequiredException ex)
                 {
                     this.errors.Add(ex);
-                    this.logger.LogDebug($"Silent auth failed, re-auth is required.\n{ex.Message}");
-                    var tokenResult = await TaskExecutor.CompleteWithin(
+                    this.logger.LogDebug($"initial {this.Name()} auth failed. Trying again with claims from exception.\n{ex.Message}");
+
+                    Func<CancellationToken, Task<TokenResult>> getTokenInteractiveWithClaims = (cancellationToken) =>
+                        this.pcaWrapper
+                            .WithPromptHint(this.promptHint)

```

Comment:
Same comment. We can probably move this as a private Func.

Reply:
Yes we could define each of these as normal funcs.

Reply:
Oh, I missed that these capture the account, so no this is a little more complicated. Trying to define a function would require either giving it access to state, which is possible, or passing the account to create a lambda function and return that. I'll play with a little.

Reply:
Yes, we can do this by passing the account and returning the Func, and I like it. I'll push shortly.